test/go-tests/test_provisioningURL.go

```diff
@@ -0,0 +1,166 @@
+          "method": "DELETE"
+        },
+        "httpResponse": {
+          "body": "Repository deleted",
+          "statusCode": 200
+        }
+      },
+    ]
+  mockserver.properties: |-
+    ###############################
+    # MockServer & Proxy Settings #
+    ###############################
+    # Socket & Port Settings
+    # socket timeout in milliseconds (default 120000)
+    mockserver.maxSocketTimeout=120000
+    # Certificate Generation
+    # dynamically generated CA key pair (if they don't already exist in
+    specified directory)
+    mockserver.dynamicallyCreateCertificateAuthorityCertificate=true
+    # save dynamically generated CA key pair in working directory
+    mockserver.directoryToSaveDynamicSSLCertificate=.
+    # certificate domain name (default "localhost")
+    mockserver.sslCertificateDomainName=localhost
+    # comma separated list of ip addresses for Subject Alternative Name domain
+    names (default empty list)
+    mockserver.sslSubjectAlternativeNameDomains=www.example.com,www.another.com
+    # comma separated list of ip addresses for Subject Alternative Name ips
+    (default empty list)
+    mockserver.sslSubjectAlternativeNameIps=127.0.0.1
+    # CORS
+    # enable CORS for MockServer REST API
+    mockserver.enableCORSForAPI=true
+    # enable CORS for all responses
+    mockserver.enableCORSForAllResponses=true
+    # Json Initialization
+    mockserver.initializationJsonPath=/config/initializerJson.json
+`
+
+// @Description: Test_ProvisioningURL tests the behaviour of the system, when no git credentials are submitted by user as
+// they should be provisioned by a 3rd partly REST API when AUTOMATIC_PROVISIONING_URL env variable is set. We are using
+// a simple mockserver to mock the responses from the 3rd partly REST API, which responses with a valid git credentials.
+// @Outcome Creating and deleting the project is tested using the provisoned git credentials
+// @Issue 7149
+func Test_ProvisioningURL(t *testing.T) {
+	projectName := "url-provisioning"
+	mockserverConfigFileName := "mockserver-config.yaml"
+	keptnNamespace := GetKeptnNameSpaceFromEnv()
+	user := GetGiteaUser()
+	url := fmt.Sprintf("http://gitea-http:3000/%s/%s", user, projectName)
+	token, _ := GetGiteaToken()
+	provisioningConfigMap := fmt.Sprintf(provisioningConfigMapTemplate, user, token, user)
+
+	mockserverconfigFilePath, err := CreateTmpFile(mockserverConfigFileName, provisioningConfigMap)
+	defer func() {
+		if err := os.Remove(mockserverconfigFilePath); err != nil {
+			t.Logf("Could not delete file: %v", err)
+		}
+	}()
+
+	t.Logf("Create mock server ConfigMap")
+	_, err = ExecuteCommandf("kubectl apply -f %s -n %s", mockserverconfigFilePath, keptnNamespace)
+	require.Nil(t, err)
+
+	t.Logf("Restarting mockserver deployment")
+	_, err = ExecuteCommandf("kubectl rollout restart deployment mockserver -n %s", keptnNamespace)
+	require.Nil(t, err)
+
+	t.Logf("Sleeping for 30s...")
+	time.Sleep(30 * time.Second)
```

Comment:
really? :(

Reply:
need to restart the deployment of the mockserver, so yes it's needed